docs/advanced/misc.rst

```diff
@@ -56,7 +76,7 @@ could be realized as follows (important changes highlighted):
 
         /* Trampoline (need one for each virtual function) */
         std::string go(int n_times) {
-            /* Acquire GIL before calling Python code */
+            /* go is called from C++, so GIL must be acquired before calling Python code */
             py::gil_scoped_acquire acquire;
```

Comment:
I believe this documentation is out of date, and I'm not sure what to do about it.

Unless I'm overlooking something, the `py::gil_scoped_acquire` here is redundant, because we have this:

https://github.com/pybind/pybind11/blob/f9f00495a3f0ef6037c215c42bd2d919590ff11f/include/pybind11/pybind11.h#L2751

I'd say it's best to delete this entire existing code block, because the main point, that the `py::gil_scoped_release` below must be matched by a `py::gil_scoped_acquire` here is no longer relevant.

When people decide to use `py::gil_scoped_release`, they are signing up for all kinds of pitfalls that they have to carefully avoid.  An outstanding pain point is when callbacks run while the Python interpreter is finalizing already (e.g. https://github.com/python/cpython/pull/28525). I'm not sure this section is the right place to go into any of that. Giving a comprehensive and accurate overview is not easy. I'd want to limit this section to only state what pybind11 does.

It would be nice to end this section with a tutorial-like reference about the GIL. On and off I've been looking for a good resource to learn more about the GIL, but I've yet to find one that I'd want to reference from here. Does anyone have suggestions?